src/pages/MainPage/MainPage.jsx

```diff
@@ -61,40 +71,70 @@ const MainPage = () => {
           </header>
           <main>
             <div className={style.box__list}>
+              <div className={style.input__box}>
+                <label>
+                  Input Value:
+                  <input type="text" value={inputValue} onChange={setInp} />
+                </label>
+                <p>Input Value: {inputValue}</p>
+              </div>
               {!users || showSkeleton ? (
                 <>
                   <SkeletonForMainpageCards cardsLength={4} />
                 </>
               ) : (
-                users.map((it, i) => {
-                  const { name, company, id,} = it;
-                  return (
-                    <div key={i} className={style.box__card}>
-                      <div className={style.box__img}>
-                        <img src={imgPerson} alt="img_bags" />
-                      </div>
-                      <div className="">{name}</div>
-                      <div className=""> Company:{company.name}</div>
+                users
+                  .filter((it) => {
+                    const address = Object.values(it.address);
+                    const company = Object.values(it.company);
+                    const allValue = Object.values(it)
+                      .concat(address)
+                      .concat(company);
+                    if (allValue.includes(inputValue)) {
+                      return it;
+                    }
+                    if (!inputValue) {
+                      return it;
+                    } 
+                  })
+                  .map((it, i) => {
+                    const { name, company, id } = it;
+                    const isFavorite = favorites.find(
+                      ({ product }) => product.name === it.name
+                    );
+                    return (
+                      <div key={i} className={style.box__card}>
+                        <div className={style.box__img}>
+                          <img src={imgPerson} alt="img_bags" />
+                        </div>
+                        <div>{name}</div>
+                        <div> Company:{company.name}</div>
 
-                      <Button
-                        className={style.button41}
-                        onClick={() => addToCart(it, i)}
-                      >
-                        Add to Project
-                      </Button>
+                        <Button
+                          className={style.button41}
+                          onClick={
+                            !isFavorite
```

Comment:
u can create a separate function `manageFavorite` and do all actios there, but not in layout